packages/block-library/src/block/edit.js

```diff
@@ -84,6 +91,40 @@ export default function ReusableBlockEdit( { attributes: { ref }, clientId } ) {
 
 	const blockProps = useBlockProps();
 
+	// local states for gudie modal
+	const [ isGudieOpen, setIsGudieOpen ] = useState( false );
+
+	// check if block has unsaved records
+	const isBlockDirty = useSelect( ( select ) => {
+		return select( coreStore ).hasEditsForEntityRecord(
+			'postType',
+			'wp_block',
+			ref
+		);
+	}, [] );
```

Comment:
Let's move this is top most `useSelect`, it's selecting from the same store anyways.

P.S. I know the term "dirty" is used a lot in this context across the codebase, but I think we can use `hasEdits` or something similar.  This is totally up to you.

Reply:
Thank you for the suggestion! 
Even I think `hasEdits` is much better. I am changing the selector to `hasEdits` and renaming the save function to `saveEditedRecords`

Reply:
Awesome.

Don't forget to move `hasEdits` selector into the top `useSelect`.

https://github.com/WordPress/gutenberg/blob/8c4282eae5882e87a518b361993ea3a8cbb90c51/packages/block-library/src/block/edit.js#L35-L55

Reply:
Added another commit! 😅  Thanks for the reminder!